pkgs/applications/audio/nuclear/default.nix

```
@@ -1,33 +1,34 @@
-{
-  appimageTools,
-  lib,
-  fetchurl,
-}: let
+{ appimageTools
+, lib
+, fetchurl
+}:
+let
   pname = "nuclear";
-  version = "0.6.27";
+  version = "0.6.30";
 
   src = fetchurl {
     url = "https://github.com/nukeop/nuclear/releases/download/v${version}/${pname}-v${version}.AppImage";
-    hash = "sha256-vCtGuId2yMVIQrMZcjN1i2buV4sah2qKupbr4LhqMbA=";
+    hash = "sha256-he1uGC1M/nFcKpMM9JKY4oeexJcnzV0ZRxhTjtJz6xw=";
   };
 
-  appimageContents = appimageTools.extract {inherit pname version src;};
+  appimageContents = appimageTools.extract { inherit pname version src; };
 in
-  appimageTools.wrapType2 {
-    inherit pname version src;
+appimageTools.wrapType2 {
+  inherit pname version src;
 
-    extraInstallCommands = ''
-      install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
-      substituteInPlace $out/share/applications/${pname}.desktop \
-        --replace 'Exec=AppRun' 'Exec=${pname}'
-      cp -r ${appimageContents}/usr/share/icons $out/share
-    '';
+  extraInstallCommands = ''
+    install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
 
-    meta = with lib; {
-      description = "Streaming music player that finds free music for you";
-      homepage = "https://nuclear.js.org/";
-      license = licenses.agpl3Plus;
-      maintainers = with maintainers; [NotAShelf ivar];
-      platforms = ["x86_64-linux"];
-    };
-  }
+  meta = with lib; {
+    description = "Streaming music player that finds free music for you";
+    homepage = "https://nuclear.js.org/";
+    license = licenses.agpl3Plus;
+    maintainers = [ maintainers.NotAShelf ];
+    platforms = [ "x86_64-linux" ];
+    mainProgram = "${pname}-${version}";
```

Weird that the binary name includes the version, create a symlink from it to `nuclear` so updates don't break and the desktop file works.

```suggestion
    mainProgram = "nuclear";
```

---

would that go in the fixup phase?

something like

```nix
fixupPhase = ''
  ln -s ${pname}-${version} ${pname}
'';
```

---

It would go in postInstall, but there are no phases in wrapType*